DokkaRoot.md

```diff
@@ -0,0 +1,197 @@
+
+### V5 Migration Guide
+
+For upgrading from V4 to V5, please refer to the [V5 Migration Guide](https://getstream.io/chat/docs/sdk/android/client/guides/chatdomain-migration/)
+
+### Changelog
+
+Check out the [changelog](https://github.com/GetStream/stream-chat-android/releases) to see the changes and improvements in each of our releases.
+
+## Getting started
+There are only a few steps to get started with Stream!
+
+* First, you need to add our dependency to the project.
+* Second, you need to set up our Client to communicate to the API.
+* Third, connect our Components to ViewModels to show data.
+* (Optional) Alternatively, you can skip using our components and build your custom UI powered by our data and persistence.
+
+Let's cover some of these steps.
+### Add dependency
+Add one of the four packages below to your dependencies for your `module/app` level `build.gradle` file:
+
+```groovy
+repositories {
+  google()
+  mavenCentral()
+  maven { url "https://jitpack.io" }
+  jcenter()
+}
+
+dependencies {
+  // Client + offline + Compose
+  implementation "io.getstream:stream-chat-android-compose:$stream_version"
+  // Client + offline + UI components
+  implementation "io.getstream:stream-chat-android-ui-components:$stream_version"
+  // Client + offline
+  implementation "io.getstream:stream-chat-android-offline:$stream_version"
+  // Client only
+  implementation "io.getstream:stream-chat-android-client:$stream_version"
+}
+
+```
+
+### Setup API Client
+
+Firstly, you need to instantiate a chat client. The Chat client will manage API call, event handling and manage the websocket connection to Stream Chat servers. You should only create the client once and re-use it across your application.
+
+```kotlin
+val apiKey = "{{ api_key }}"
+val token = "{{ chat_user_token }}"
+
+val client = ChatClient.Builder(apiKey, applicationContext).build()
+```
+
+Secondly you need authenticate and connect the user.
+```kotlin
+val user = User(
+    id = "summer-brook-2",
+    extraData = mutableMapOf(
+        "name" to "Paranoid Android",
+        "image" to "https://bit.ly/2TIt8NR",
+    ),
+)
+client.connectUser(
+    user = user,
+    token = token, // or client.devToken(userId); if auth is disabled for your app
+).enqueue { result ->
+    if (result.isSuccess) {
+        // Handle success
+    } else {
+        // Handler error
+    }
+}
+```
+
+The user token is typically provided by your backend when you login or register in the app. If authentication is disabled for your app, you can also use a `ChatClient#devToken` to generate an insecure token for development. Of course, you should never launch into production with authentication disabled.
+
+For more complex token generation and expiration examples, have a look at [Token Expiration](https://getstream.io/chat/docs/android/tokens_and_authentication/#token-expiration).
```

Comment:
What do you think about mentioning the [Token Generator](https://getstream.io/chat/docs/react/token_generator/)?
Many new users have asked me about how to generate and test authentication tokens without their backend service.